sklearn/utils/tests/test_array_api.py

```diff
@@ -371,3 +396,34 @@ def test_get_namespace_array_api_isdtype(wrapper):
 
     with pytest.raises(ValueError, match="Unrecognized data type"):
         assert xp.isdtype(xp.int16, "unknown")
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
```

Comment:
I think we should test with non-float dtypes here. Maybe int32, uint32, int64 and uint64.

It doesn't hurt to also test on float32 / float64 but I don't think this function will be used on floating point data in practice.